src/sgame/sg_bot_local.h

```diff
@@ -115,12 +116,19 @@ struct botMemory_t
 
 	int         futureAimTime;
 	int         futureAimTimeInterval;
-	vec3_t      futureAim;
-	botNavCmd_t nav;
+	glm::vec3   futureAim;
+private:
+	botNavCmd_t m_nav;
+public:
+	botNavCmd_t const& nav( void ) const { return m_nav; };
+	void clearNav( void ) { memset( &m_nav, 0, sizeof( m_nav ) ); }
+	void setGoal( botTarget_t target, bool direct ) { goal = target; m_nav.directPathToGoal = direct; }
+	friend void G_BotThink( gentity_t * );
```

Comment:
This friend stuff is very ugly. Maybe if it were going to some sort of sensible end goal but I doubt it...

Reply:
I think the idea is to identify the functions which use it and have a easier time to check they are correct. and progressively migrate to correct setters or smth.

It's not my code so I don't really know though